File: src/util/entities/ApplicationCommand.ts
/*
	Spacebar: A FOSS re-implementation and extension of the Discord.com backend.
	Copyright (C) 2023 Spacebar and Spacebar Contributors

	This program is free software: you can redistribute it and/or modify
	it under the terms of the GNU Affero General Public License as published
	by the Free Software Foundation, either version 3 of the License, or
	(at your option) any later version.

	This program is distributed in the hope that it will be useful,
	but WITHOUT ANY WARRANTY; without even the implied warranty of
	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
	GNU Affero General Public License for more details.

	You should have received a copy of the GNU Affero General Public License
	along with this program.  If not, see <https://www.gnu.org/licenses/>.
*/

import { Column, Entity } from "typeorm";
import { BaseClass } from "./BaseClass";
import {
	ApplicationCommandHandlerType,
	ApplicationCommandOption,
	ApplicationCommandIndexPermissions,
	ApplicationCommandType,
	Snowflake,
	ApplicationIntegrationType,
	InteractionContextType,
} from "@spacebar/schemas";

@Entity({
	name: "application_commands",
})
export class ApplicationCommand extends BaseClass {
	@Column({ default: ApplicationCommandType.CHAT_INPUT })
	type?: ApplicationCommandType;

	@Column()
	application_id: Snowflake;

	@Column({ nullable: true })
	guild_id?: Snowflake;

	@Column()
	name: string;

	@Column({ nullable: true, type: "simple-json" })
	name_localizations?: Record<string, string>;

	@Column()
	description: string;

	@Column({ nullable: true, type: "simple-json" })
	description_localizations?: Record<string, string>;

	@Column({ type: "simple-json", default: [] })
	options?: ApplicationCommandOption[];

	@Column({ nullable: true, type: String })
	default_member_permissions: string | null;

	/*
	 * @deprecated
	 */
	@Column({ default: true })
	dm_permission?: boolean;

	@Column({ nullable: true, type: "simple-json" })
	permissions?: ApplicationCommandIndexPermissions;

	@Column({ default: false })
	nsfw?: boolean;

	@Column({ type: "simple-json", default: [] })
	integration_types?: ApplicationIntegrationType[];

	@Column({ default: 0 })
	global_popularity_rank?: number;

	@Column({ type: "simple-json", default: [] })
	contexts?: InteractionContextType[];

	@Column({ default: 0 })
	version: Snowflake;

	@Column({ default: 0 })
	handler?: ApplicationCommandHandlerType;
}

File: extra/admin-api/Spacebar.Db/Models/ApplicationCommand.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Spacebar.Db.Models;

[Table("application_commands")]
public partial class ApplicationCommand
{
    [Key]
    [Column("id", TypeName = "character varying")]
    public string Id { get; set; } = null!;

    [Column("type")]
    public int Type { get; set; }

    [Column("application_id", TypeName = "character varying")]
    public string ApplicationId { get; set; } = null!;

    [Column("guild_id", TypeName = "character varying")]
    public string? GuildId { get; set; }

    [Column("name", TypeName = "character varying")]
    public string Name { get; set; } = null!;

    [Column("name_localizations")]
    public string? NameLocalizations { get; set; }

    [Column("description", TypeName = "character varying")]
    public string Description { get; set; } = null!;

    [Column("description_localizations")]
    public string? DescriptionLocalizations { get; set; }

    [Column("options")]
    public string Options { get; set; } = null!;

    [Column("default_member_permissions", TypeName = "character varying")]
    public string? DefaultMemberPermissions { get; set; }

    [Column("dm_permission")]
    public bool DmPermission { get; set; }

    [Column("permissions")]
    public string? Permissions { get; set; }

    [Column("nsfw")]
    public bool Nsfw { get; set; }

    [Column("integration_types")]
    public string? IntegrationTypes { get; set; }

    [Column("global_popularity_rank")]
    public int GlobalPopularityRank { get; set; }

    [Column("contexts")]
    public string? Contexts { get; set; }

    [Column("version", TypeName = "character varying")]
    public string Version { get; set; } = null!;

    [Column("handler")]
    public int Handler { get; set; }
}
